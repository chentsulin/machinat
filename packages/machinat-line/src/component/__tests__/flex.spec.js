import {
  Flex,
  FlexBox,
  FlexButton,
  FlexFiller,
  FlexIcon,
  FlexImage,
  FlexSeparator,
  FlexSpacer,
  FlexText,
  FlexHeader,
  FlexHero,
  FlexBody,
  FlexFooter,
  FlexBubbleContainer,
  FlexCarouselContainer,
  FlexMessage,
} from '../flex';
