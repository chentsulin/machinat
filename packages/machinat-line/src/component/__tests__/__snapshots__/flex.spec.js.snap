// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News flex sample match snapshot 1`] = `
Array [
  Object {
    "altText": undefined,
    "contents": Object {
      "body": Object {
        "action": undefined,
        "contents": Array [
          Object {
            "action": undefined,
            "contents": Array [
              Object {
                "action": undefined,
                "align": undefined,
                "aspectMode": "cover",
                "aspectRatio": "4:3",
                "backgroundColor": undefined,
                "flex": undefined,
                "gravity": "bottom",
                "margin": undefined,
                "size": "sm",
                "type": "image",
                "url": "https://scdn.line-apps.com/n/channel_devcenter/img/fx/02_1_news_thumbnail_1.png",
              },
              Object {
                "action": undefined,
                "align": undefined,
                "aspectMode": "cover",
                "aspectRatio": "4:3",
                "backgroundColor": undefined,
                "flex": undefined,
                "gravity": undefined,
                "margin": "md",
                "size": "sm",
                "type": "image",
                "url": "https://scdn.line-apps.com/n/channel_devcenter/img/fx/02_1_news_thumbnail_2.png",
              },
            ],
            "flex": 1,
            "layout": "vertical",
            "margin": undefined,
            "spacing": undefined,
            "type": "box",
          },
          Object {
            "action": undefined,
            "contents": Array [
              Object {
                "action": undefined,
                "align": undefined,
                "color": undefined,
                "flex": 1,
                "gravity": "top",
                "margin": undefined,
                "maxLines": undefined,
                "size": "xs",
                "text": "7 Things to Know for Today",
                "type": "text",
                "weight": undefined,
                "wrap": undefined,
              },
              Object {
                "color": undefined,
                "margin": undefined,
                "type": "separator",
              },
              Object {
                "action": undefined,
                "align": undefined,
                "color": undefined,
                "flex": 2,
                "gravity": "top",
                "margin": undefined,
                "maxLines": undefined,
                "size": "xs",
                "text": "Hay fever goes wild",
                "type": "text",
                "weight": undefined,
                "wrap": undefined,
              },
              Object {
                "color": undefined,
                "margin": undefined,
                "type": "separator",
              },
              Object {
                "action": undefined,
                "align": undefined,
                "color": undefined,
                "flex": 2,
                "gravity": "top",
                "margin": undefined,
                "maxLines": undefined,
                "size": "xs",
                "text": "LINE Pay Begins Barcode Payment Service",
                "type": "text",
                "weight": undefined,
                "wrap": undefined,
              },
              Object {
                "color": undefined,
                "margin": undefined,
                "type": "separator",
              },
              Object {
                "action": undefined,
                "align": undefined,
                "color": undefined,
                "flex": 1,
                "gravity": "top",
                "margin": undefined,
                "maxLines": undefined,
                "size": "xs",
                "text": "LINE Adds LINE Wallet",
                "type": "text",
                "weight": undefined,
                "wrap": undefined,
              },
            ],
            "flex": 2,
            "layout": "vertical",
            "margin": undefined,
            "spacing": undefined,
            "type": "box",
          },
        ],
        "flex": undefined,
        "layout": "horizontal",
        "margin": undefined,
        "spacing": "md",
        "type": "box",
      },
      "direction": "ltr",
      "footer": Object {
        "action": undefined,
        "contents": Array [
          Object {
            "action": Object {
              "label": undefined,
              "type": "uri",
              "uri": "http://linecorp.com/",
            },
            "color": undefined,
            "flex": undefined,
            "gravity": undefined,
            "height": undefined,
            "margin": undefined,
            "style": undefined,
            "type": "button",
          },
        ],
        "flex": undefined,
        "layout": "horizontal",
        "margin": undefined,
        "spacing": undefined,
        "type": "box",
      },
      "header": Object {
        "action": undefined,
        "contents": Array [
          Object {
            "action": undefined,
            "align": undefined,
            "color": "#aaaaaa",
            "flex": undefined,
            "gravity": undefined,
            "margin": undefined,
            "maxLines": undefined,
            "size": "sm",
            "text": "NEWS DIGEST",
            "type": "text",
            "weight": "bold",
            "wrap": undefined,
          },
        ],
        "flex": undefined,
        "layout": "horizontal",
        "margin": undefined,
        "spacing": undefined,
        "type": "box",
      },
      "hero": Object {
        "action": Object {
          "label": undefined,
          "type": "uri",
          "uri": "http://linecorp.com/",
        },
        "align": undefined,
        "aspectMode": "cover",
        "aspectRatio": "20:13",
        "backgroundColor": undefined,
        "flex": undefined,
        "gravity": undefined,
        "margin": undefined,
        "size": "full",
        "type": "image",
        "url": "https://scdn.line-apps.com/n/channel_devcenter/img/fx/01_4_news.png",
      },
      "type": "bubble",
    },
    "type": "flex",
  },
]
`;

exports[`Restaurant flex sample match snapshot 1`] = `
Array [
  Object {
    "altText": undefined,
    "contents": Object {
      "body": Object {
        "action": undefined,
        "contents": Array [
          Object {
            "action": undefined,
            "align": undefined,
            "color": undefined,
            "flex": undefined,
            "gravity": undefined,
            "margin": undefined,
            "maxLines": undefined,
            "size": "xl",
            "text": "Brown Cafe",
            "type": "text",
            "weight": "bold",
            "wrap": undefined,
          },
          Object {
            "action": undefined,
            "contents": Array [
              Object {
                "aspectRatio": undefined,
                "margin": undefined,
                "size": "sm",
                "type": "icon",
                "url": "https://scdn.line-apps.com/n/channel_devcenter/img/fx/review_gold_star_28.png",
              },
              Object {
                "aspectRatio": undefined,
                "margin": undefined,
                "size": "sm",
                "type": "icon",
                "url": "https://scdn.line-apps.com/n/channel_devcenter/img/fx/review_gold_star_28.png",
              },
              Object {
                "aspectRatio": undefined,
                "margin": undefined,
                "size": "sm",
                "type": "icon",
                "url": "https://scdn.line-apps.com/n/channel_devcenter/img/fx/review_gold_star_28.png",
              },
              Object {
                "aspectRatio": undefined,
                "margin": undefined,
                "size": "sm",
                "type": "icon",
                "url": "https://scdn.line-apps.com/n/channel_devcenter/img/fx/review_gold_star_28.png",
              },
              Object {
                "aspectRatio": undefined,
                "margin": undefined,
                "size": "sm",
                "type": "icon",
                "url": "https://scdn.line-apps.com/n/channel_devcenter/img/fx/review_gray_star_28.png",
              },
              Object {
                "action": undefined,
                "align": undefined,
                "color": "#999999",
                "flex": 0,
                "gravity": undefined,
                "margin": "md",
                "maxLines": undefined,
                "size": "sm",
                "text": "4.0",
                "type": "text",
                "weight": undefined,
                "wrap": undefined,
              },
              Object {
                "type": "filler",
              },
            ],
            "flex": undefined,
            "layout": "baseline",
            "margin": "md",
            "spacing": undefined,
            "type": "box",
          },
          Object {
            "action": undefined,
            "contents": Array [
              Object {
                "action": undefined,
                "contents": Array [
                  Object {
                    "action": undefined,
                    "align": undefined,
                    "color": "#aaaaaa",
                    "flex": 1,
                    "gravity": undefined,
                    "margin": undefined,
                    "maxLines": undefined,
                    "size": "sm",
                    "text": "Place",
                    "type": "text",
                    "weight": undefined,
                    "wrap": undefined,
                  },
                  Object {
                    "action": undefined,
                    "align": undefined,
                    "color": "#666666",
                    "flex": 5,
                    "gravity": undefined,
                    "margin": undefined,
                    "maxLines": undefined,
                    "size": "sm",
                    "text": "Miraina Tower, 4-1-6 Shinjuku, Tokyo",
                    "type": "text",
                    "weight": undefined,
                    "wrap": true,
                  },
                ],
                "flex": undefined,
                "layout": "baseline",
                "margin": undefined,
                "spacing": "sm",
                "type": "box",
              },
              Object {
                "action": undefined,
                "contents": Array [
                  Object {
                    "action": undefined,
                    "align": undefined,
                    "color": "#aaaaaa",
                    "flex": 1,
                    "gravity": undefined,
                    "margin": undefined,
                    "maxLines": undefined,
                    "size": "sm",
                    "text": "Time",
                    "type": "text",
                    "weight": undefined,
                    "wrap": undefined,
                  },
                  Object {
                    "action": undefined,
                    "align": undefined,
                    "color": "#666666",
                    "flex": 5,
                    "gravity": undefined,
                    "margin": undefined,
                    "maxLines": undefined,
                    "size": "sm",
                    "text": "10:00 - 23:00",
                    "type": "text",
                    "weight": undefined,
                    "wrap": true,
                  },
                ],
                "flex": undefined,
                "layout": "baseline",
                "margin": undefined,
                "spacing": "sm",
                "type": "box",
              },
            ],
            "flex": undefined,
            "layout": "vertical",
            "margin": "lg",
            "spacing": "sm",
            "type": "box",
          },
        ],
        "flex": undefined,
        "layout": "vertical",
        "margin": undefined,
        "spacing": undefined,
        "type": "box",
      },
      "direction": "ltr",
      "footer": Object {
        "action": undefined,
        "contents": Array [
          Object {
            "action": Object {
              "label": "CALL",
              "type": "uri",
              "uri": "https://linecorp.com",
            },
            "color": undefined,
            "flex": undefined,
            "gravity": undefined,
            "height": "sm",
            "margin": undefined,
            "style": "link",
            "type": "button",
          },
          Object {
            "action": Object {
              "label": "WEBSITE",
              "type": "uri",
              "uri": "https://linecorp.com",
            },
            "color": undefined,
            "flex": undefined,
            "gravity": undefined,
            "height": "sm",
            "margin": undefined,
            "style": "link",
            "type": "button",
          },
          Object {
            "size": undefined,
            "type": "spacer",
          },
        ],
        "flex": 0,
        "layout": "vertical",
        "margin": undefined,
        "spacing": "sm",
        "type": "box",
      },
      "hero": Object {
        "action": Object {
          "label": undefined,
          "type": "uri",
          "uri": "http://linecorp.com/",
        },
        "align": undefined,
        "aspectMode": "cover",
        "aspectRatio": "20:13",
        "backgroundColor": undefined,
        "flex": undefined,
        "gravity": undefined,
        "margin": undefined,
        "size": "full",
        "type": "image",
        "url": "https://scdn.line-apps.com/n/channel_devcenter/img/fx/01_1_cafe.png",
      },
      "type": "bubble",
    },
    "type": "flex",
  },
]
`;

exports[`Shopping flex sample match snapshot 1`] = `
Array [
  Object {
    "altText": undefined,
    "contents": Object {
      "contents": Array [
        Object {
          "body": Object {
            "action": undefined,
            "contents": Array [
              Object {
                "action": undefined,
                "align": undefined,
                "color": undefined,
                "flex": undefined,
                "gravity": undefined,
                "margin": undefined,
                "maxLines": undefined,
                "size": "xl",
                "text": "Arm Chair, White",
                "type": "text",
                "weight": "bold",
                "wrap": true,
              },
              Object {
                "action": undefined,
                "contents": Array [
                  Object {
                    "action": undefined,
                    "align": undefined,
                    "color": undefined,
                    "flex": 0,
                    "gravity": undefined,
                    "margin": undefined,
                    "maxLines": undefined,
                    "size": "xl",
                    "text": "$49",
                    "type": "text",
                    "weight": "bold",
                    "wrap": true,
                  },
                  Object {
                    "action": undefined,
                    "align": undefined,
                    "color": undefined,
                    "flex": 0,
                    "gravity": undefined,
                    "margin": undefined,
                    "maxLines": undefined,
                    "size": "sm",
                    "text": ".99",
                    "type": "text",
                    "weight": "bold",
                    "wrap": true,
                  },
                ],
                "flex": undefined,
                "layout": undefined,
                "margin": undefined,
                "spacing": undefined,
                "type": "box",
              },
            ],
            "flex": undefined,
            "layout": "vertical",
            "margin": undefined,
            "spacing": "sm",
            "type": "box",
          },
          "direction": "ltr",
          "footer": Object {
            "action": undefined,
            "contents": Array [
              Object {
                "action": Object {
                  "label": "Add to Cart",
                  "type": "uri",
                  "uri": "https://linecorp.com",
                },
                "color": undefined,
                "flex": undefined,
                "gravity": undefined,
                "height": undefined,
                "margin": undefined,
                "style": "primary",
                "type": "button",
              },
              Object {
                "action": Object {
                  "label": "Add to wishlist",
                  "type": "uri",
                  "uri": "https://linecorp.com",
                },
                "color": undefined,
                "flex": undefined,
                "gravity": undefined,
                "height": undefined,
                "margin": undefined,
                "style": undefined,
                "type": "button",
              },
            ],
            "flex": undefined,
            "layout": "vertical",
            "margin": undefined,
            "spacing": "sm",
            "type": "box",
          },
          "hero": Object {
            "action": undefined,
            "align": undefined,
            "aspectMode": "cover",
            "aspectRatio": "20:13",
            "backgroundColor": undefined,
            "flex": undefined,
            "gravity": undefined,
            "margin": undefined,
            "size": "full",
            "type": "image",
            "url": "https://scdn.line-apps.com/n/channel_devcenter/img/fx/01_5_carousel.png",
          },
          "type": "bubble",
        },
        Object {
          "body": Object {
            "action": undefined,
            "contents": Array [
              Object {
                "action": undefined,
                "align": undefined,
                "color": undefined,
                "flex": undefined,
                "gravity": undefined,
                "margin": undefined,
                "maxLines": undefined,
                "size": "xl",
                "text": "Metal Desk Lamp",
                "type": "text",
                "weight": "bold",
                "wrap": true,
              },
              Object {
                "action": undefined,
                "contents": Array [
                  Object {
                    "action": undefined,
                    "align": undefined,
                    "color": undefined,
                    "flex": 0,
                    "gravity": undefined,
                    "margin": undefined,
                    "maxLines": undefined,
                    "size": "xl",
                    "text": "$11",
                    "type": "text",
                    "weight": "bold",
                    "wrap": true,
                  },
                  Object {
                    "action": undefined,
                    "align": undefined,
                    "color": undefined,
                    "flex": 0,
                    "gravity": undefined,
                    "margin": undefined,
                    "maxLines": undefined,
                    "size": "sm",
                    "text": ".99",
                    "type": "text",
                    "weight": "bold",
                    "wrap": true,
                  },
                ],
                "flex": undefined,
                "layout": undefined,
                "margin": undefined,
                "spacing": undefined,
                "type": "box",
              },
              Object {
                "action": undefined,
                "align": undefined,
                "color": "#ff5551",
                "flex": 0,
                "gravity": undefined,
                "margin": "md",
                "maxLines": undefined,
                "size": "xxs",
                "text": "Temporarily out of stock",
                "type": "text",
                "weight": undefined,
                "wrap": true,
              },
            ],
            "flex": undefined,
            "layout": "vertical",
            "margin": undefined,
            "spacing": "sm",
            "type": "box",
          },
          "direction": "ltr",
          "footer": Object {
            "action": undefined,
            "contents": Array [
              Object {
                "action": Object {
                  "label": "Add to Cart",
                  "type": "uri",
                  "uri": "https://linecorp.com",
                },
                "color": "#aaaaaa",
                "flex": undefined,
                "gravity": undefined,
                "height": undefined,
                "margin": undefined,
                "style": "primary",
                "type": "button",
              },
              Object {
                "action": Object {
                  "label": "Add to wishlist",
                  "type": "uri",
                  "uri": "https://linecorp.com",
                },
                "color": undefined,
                "flex": undefined,
                "gravity": undefined,
                "height": undefined,
                "margin": undefined,
                "style": undefined,
                "type": "button",
              },
            ],
            "flex": undefined,
            "layout": "vertical",
            "margin": undefined,
            "spacing": "sm",
            "type": "box",
          },
          "hero": Object {
            "action": undefined,
            "align": undefined,
            "aspectMode": "cover",
            "aspectRatio": "20:13",
            "backgroundColor": undefined,
            "flex": undefined,
            "gravity": undefined,
            "margin": undefined,
            "size": "full",
            "type": "image",
            "url": "https://scdn.line-apps.com/n/channel_devcenter/img/fx/01_6_carousel.png",
          },
          "type": "bubble",
        },
        Object {
          "body": Object {
            "action": undefined,
            "contents": Array [
              Object {
                "action": Object {
                  "label": "See more",
                  "type": "uri",
                  "uri": "https://linecorp.com",
                },
                "color": undefined,
                "flex": 1,
                "gravity": "center",
                "height": undefined,
                "margin": undefined,
                "style": undefined,
                "type": "button",
              },
            ],
            "flex": undefined,
            "layout": "vertical",
            "margin": undefined,
            "spacing": "sm",
            "type": "box",
          },
          "direction": "ltr",
          "type": "bubble",
        },
      ],
      "type": "carousel",
    },
    "type": "flex",
  },
]
`;
